### DISCUSSION TITLE
Query Headers Caching

### DISCUSSION THREAD

**Discussion by michailozdemir** [TanStack/query#8421] [Q&A]
I have a simple question I didn't find answer to. Let's imagine some token refresh functionality which requires a query retry after the token has been re-fetched. In order to have a successful query retry, I need to update the headers for that query.

For some reason the generic axios headers update doesn't help and I can assume that query is being cached and so its headers. 
What is the proper way to update the headers for query in such case and is there any way at all?

**Comment by TkDodo** [comment]
since you’re using axios, I’d suggest an axios request/response interceptor to add the header and re-fetch the token.
